builtin/providers/aws/resource_aws_api_gateway_integration.go

```diff
@@ -0,0 +1,185 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayIntegration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayIntegrationCreate,
+		Read:   resourceAwsApiGatewayIntegrationRead,
+		Update: resourceAwsApiGatewayIntegrationUpdate,
+		Delete: resourceAwsApiGatewayIntegrationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"http_method": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"uri": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"credentials": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"integration_http_method": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"request_templates": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayIntegrationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	var integrationHttpMethod *string
+	if v, ok := d.GetOk("integration_http_method"); ok {
+		integrationHttpMethod = aws.String(v.(string))
+	}
+	var uri *string
+	if v, ok := d.GetOk("uri"); ok {
+		uri = aws.String(v.(string))
+	}
+	templates := make(map[string]string)
+	for k, v := range d.Get("request_templates").(map[string]interface{}) {
+		templates[k] = v.(string)
+	}
+
+	var credentials *string
+	if val, ok := d.GetOk("credentials"); ok {
+		credentials = aws.String(val.(string))
+	}
+
+	_, err := conn.PutIntegration(&apigateway.PutIntegrationInput{
+		HttpMethod: aws.String(d.Get("http_method").(string)),
+		ResourceId: aws.String(d.Get("resource_id").(string)),
+		RestApiId:  aws.String(d.Get("api_id").(string)),
+		Type:       aws.String(d.Get("type").(string)),
+		IntegrationHttpMethod: integrationHttpMethod,
+		Uri:                uri,
+		RequestParameters:  nil,
```

Comment:
I see. :cry: In that case can you at least leave a comment here? e.g.

> Not implemented due to `https://github.com/hashicorp/terraform/issues/2143`

So that once we get that bug fixed, we can get back to this and finish the implementation?